viewing_party/party.py

```diff
@@ -1,23 +1,193 @@
+        if title in user_data["watchlist"][i].values(): #add to watched list
+            user_data["watched"].append(user_data["watchlist"][i]) 
+        else: #remain in watchlist
+            new_watchlist.append(user_data["watchlist"][i])
+
+    user_data["watchlist"] = new_watchlist #update watchlist in user data
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+
+    #initiate running total for rating score
+    total = 0
+    for i in range(len(user_data["watched"])):
+            total += user_data["watched"][i]["rating"]
+    
+    try: #get avg rating score
+        average = total/len(user_data["watched"])
+    except ZeroDivisionError: #if empty list
+        return 0
+
+    return average
+
+def get_most_watched_genre(user_data):
 
+    #return None for empty watched list
+    if len(user_data["watched"]) == 0:
+        return None
+    
+    #set count for each genre
+    genre_count = {"Fantasy": 0, "Action": 0, "Intrigue": 0, "Horror": 0}
 
+    #loop over each item's genre and check which genre it matches, 
+    for movie in user_data["watched"]:
+        if movie["genre"] == "Fantasy":
+            genre_count["Fantasy"]+=1
+        elif movie["genre"] == "Action":
+            genre_count["Action"]+=1
+        elif movie["genre"] == "Intrigue":
+            genre_count["Intrigue"] += 1
+        else:
+            genre_count["Horror"] += 1
+    
+    #return max genre count genre
+    return max(genre_count, key = genre_count.get)
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    #user_data= {watched: [{}, {}, {}], friends: [watched: [{title: string}, {title: string}, {title: string}], [], []]}
+
+    #compare movie titles for every movie in user_data["watched"] to movie titles for every movie of every friend in user_data["friends"] and append unique MOVIES to list
+    
+    #access friends' movie titles as list of strings
+    friends_watched_titles = []
+
+    for friends in user_data["friends"]:
+        for movie in friends["watched"]:
+            friends_watched_titles.append(movie["title"])
+
+    #initialize unique movie list of dicts
+    unique_movies = []   
+
+    #compare every title between user and friends
+    for user_movie in user_data["watched"]:
+        if user_movie["title"] not in friends_watched_titles: #compare title names
+            unique_movies.append(user_movie) #append unique movies
+
+    #list of dictionaries
+    return unique_movies
+
+
+def get_friends_unique_watched(user_data):
+    
+    #get list of dicts of all friend watched movies
+    friends_watched_movies = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_movies.append(movie)

```

Comment:
We repeat this nested for loop to get the friend's movies titles throughout this project. To DRY up our code, we can put this block of code in a helper function.

Reply:
Can we review this in our one-on-one? I do not see this nested loop repeated in my code.